pages/home.jsx

```diff
@@ -35,27 +35,36 @@ var FeaturedPost = React.createClass({
   propTypes: {
     data: React.PropTypes.object.isRequired
   },
+  componentWillUpdate: function(nextProps, nextState) {
+    this.parsedMomentDate = moment(new Date(nextProps.data.publishedDate));
```

Comment:
Hmm, I recommend against storing variables that are used by our rendering logic directly on `this`. I *think* it's ok to store e.g. the return value of `window.setTimeout` on `this` because those values don't affect the rendering of our component, but otherwise we really want to use `this.props` and `this.state` as much as possible, because then React takes care of automatically re-rendering things for us (it's not a *huge* deal for this particular component, but as components become more complex, it will help keep things conceptually understandable and prevent bugs).

Also, assuming that a valid published date will *always* exist on the first component update seems fragile to me; why not just check during `render` to see if `props.data.publishedDate` is truthy or non-empty? It's not actually bad to do some view-related computation during `render`, especially if we use [`PureRenderMixin`](https://facebook.github.io/react/docs/pure-render-mixin.html) to speed things up so that re-renders are performed as rarely as possible. And it keeps the behavior of our components extremely simple and easy to understand.